Content.Server/Dice/DiceSystem.cs

```diff
@@ -22,7 +26,38 @@ public override void Roll(EntityUid uid, DiceComponent? die = null)
         var roll = _random.Next(1, die.Sides + 1);
         SetCurrentSide(uid, roll, die);
 
-        _popup.PopupEntity(Loc.GetString("dice-component-on-roll-land", ("die", uid), ("currentSide", die.CurrentValue)), uid);
         _audio.PlayPvs(die.Sound, uid);
+
+        if (die.DiceBomb && !TryComp<ExplodeOnTriggerComponent>(uid, out var explodeTrigger)) // True only on first roll, so cannot be activated twice
```

Comment:
I think DiceBomb should be a separate system, not built into an existing DiceSystem. Just make the events you need and subscribe to them in a separate file.